dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -139,7 +139,7 @@
 public class UserResource implements AuthenticatedResourceInterface, SourceControlResourceInterface {
     protected static final Pattern GITHUB_ID_PATTERN = Pattern.compile(".*/u/(\\d+).*");
     private static final Logger LOG = LoggerFactory.getLogger(UserResource.class);
-    private static final Pattern USERNAME_CONTAINS_KEYWORD_PATTERN = Pattern.compile("(?i)(dockstore|admin|curator|system|manager)");
+    protected static final Pattern USERNAME_CONTAINS_KEYWORD_PATTERN = Pattern.compile("(?i)(dockstore|admin|curator|system|manager)");
```

Comment:
I'm not sure why, but I don't like having dependencies between Resource files (you made this protected so TokenResource can access it). And even though this regex is simple enough that even I can understand it without having to look anything up, it would still be good to have unit tests, especially if it evolves (maybe it needs to catch d0ckstore, maybe the test becomes more complex than a single regex can handle, etc.).

So I would suggest a Helper class (we have that pattern), where you could create unit tests against it. And have a method that returns banned keywords for constructing the error message, etc.

But your code works and we don't want to prematurely architect, so I'm good with it.